src/key_io.cpp

```diff
@@ -68,11 +68,18 @@ class DestinationEncoder : public boost::static_visitor<std::string>
     std::string operator()(const CNoDestination& no) const { return {}; }
 };
 
-CTxDestination DecodeDestination(const std::string& str, const CChainParams& params)
+CTxDestination DecodeDestination(const std::string& str, const CChainParams& params, ColorIdentifier* colorId = nullptr)
 {
     std::vector<unsigned char> data;
     uint160 hash;
     if (DecodeBase58Check(str, data)) {
+
+        if (data[35] == OP_COLOR) {
```

Comment:
It is better to use the existing decision logic for colored coin like `CScript::IsColoredScript()`, as it is better to have a single logic to determine them.

Reply:
ok